src/main/java/com/commercetools/sync/commons/helpers/BaseReferenceResolver.java

```diff
@@ -41,24 +41,24 @@ protected BaseReferenceResolver(@Nonnull final S options) {
     public abstract CompletionStage<T> resolveReferences(@Nonnull T draft);
 
     /**
-     * This method fetches the id value on the passed {@link ResourceIdentifier}, if valid. If it is not valid, a
+     * This method fetches the key value on the passed {@link Reference}, if valid. If it is not valid, a
      * {@link ReferenceResolutionException} will be thrown. The validity checks are:
      * <ul>
      * <li>Checks if the id value is not null or not empty.</li>
      * </ul>
      * If the above checks pass, the key value is returned. Otherwise a {@link ReferenceResolutionException} is thrown.
      *
-     * @param resourceIdentifier the reference from which the key value is validated and returned.
+     * @param reference the reference from which the key value is validated and returned.
      * @return the Id value on the {@link ResourceIdentifier}
      * @throws ReferenceResolutionException if any of the validation checks fail.
      */
-    @Nonnull // todo (ahmetoz) could be better to rename this to getKeyFromReference and type to Reference<T>
-    protected static String getIdFromResourceIdentifier(@Nonnull final ResourceIdentifier resourceIdentifier)
+    @Nonnull
+    protected static <T> String getKeyFromReference(@Nonnull final Reference<T> reference)
```

Comment:
It's only used for references and the parameter is not a resource identifier anymore, so I would rename it reference -> `getIdFromReference` for me that does not explain the method because the usage of it to get a key from reference so I renamed that way.

Reply:
what do you think of "getIdAsKeyFromReference"

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/553/commits/a83a82b1796d4468147625f0275d08b24a79ca47

Reply:
thanks